notebooks/working/Simple_Simulation.md

```diff
@@ -0,0 +1,382 @@
+```
+
+```python
+## Regress travel time on travel distance
+
+time_on_distance_reg = fit_regression(X=['total_travel_distance'], 
+                                      y = ['total_travel_time'], data=drive_alone_df, plotting = True)
+
+time_on_distance_reg.summary()
+```
+
+### Simulate travel time and cost based on the distributional regression
+
+```python
+## residuals spread -- assuming homoscedasticity
+time_on_dist_std = time_on_distance_reg.resid.std()
+cost_on_dist_std = cost_on_distance_reg.resid.std()
+
+sample_size = len(drive_alone_df)
+```
+
+```python
+total_travel_time_sim = ( time_on_distance_reg.params[0] + 
+                         time_on_distance_reg.params[1]*drive_alone_df['total_travel_distance']
+                        + np.random.normal(loc=0, scale=time_on_dist_std, size = sample_size) )
+
+total_travel_cost_sim = ( cost_on_distance_reg.params[0] + 
+                         cost_on_distance_reg.params[1]*drive_alone_df['total_travel_distance']
+                        + np.random.normal(loc=0, scale=cost_on_dist_std, size = sample_size) )
+```
+
+### Create a simulation dataframe
+
+```python
+simulation_df = drive_alone_df.copy()
+simulation_df['total_travel_time'] = total_travel_time_sim
+simulation_df['total_travel_cost'] = total_travel_cost_sim
+```
+
+### Simulate the utility equation based on hypothetical parameters
+
+```python
+simulation_df.columns
+```
+
+```python
+## Assumed coefficients for above columns
+coeffs = np.array([-.5, -1.5, 3, 5, 2, -7 ])
+asc_drive = 6.
+```
+
+```python
+utilities = asc_drive + np.dot( simulation_df, coeffs) + np.random.normal(loc=0, scale = 1, size = sample_size)
+
+simulation_df['utilities'] = utilities
+```
+
+## Estimation
+
+Note that here, I'm treating the utilities as an observed quantity that I'm trying to estimate. This will get more complicated as we include different modes and actually try to maximize the correct likelihood function. 
+
+The thing I need to point out here is that irrespective of our causal graph, we will always recover the paramters in coeffs (defined above) when we run the regression. The question becomes, are the coefficients on each of the variables (the true coefficients) causal? That depends on the causal graph:
+
+- In the case of independent variables, the true causal effect of distance is the 0.1, the same value in the coeffs array. Thus, running a regression on all the variables in this case would return the true causal estimate. 
+
+- In the case where travel cost and travel time are descendents of distance, the true causal effect of distance becomes: 0.1 - 0.5*(1.28) - 1.5*(0.22) = -0.87. We will only recover this value if we omit travel distance and travel time from the utility equation. Alternatively, we can keep them in the equation, but then don't assign the coefficient for distance any causal interpretation, and post-process the results (and make parametric assumptions!) to calculate the true causal effect. 
```

Comment:
These numbers no longer match the example shown in the code though I imagine they once did.